Xamarin.Forms.Core/PlatformConfiguration/iOSSpecific/VisualElement.cs

```diff
@@ -30,6 +31,161 @@ public static BlurEffectStyle GetBlurEffect(this IPlatformElementConfiguration<i
 			return config;
 		}
 
+		#region Shadow Settings
+
+		public class ShadowEffect : RoutingEffect
+		{
+			public ShadowEffect() : base("Xamarin.ShadowEffect") { }
+		}
+
+		public static readonly BindableProperty IsShadowEnabledProperty =
+			BindableProperty.Create("IsShadowEnabled", typeof(bool),
+			typeof(VisualElement), false, propertyChanged: OnIsShadowEnabledChanged);
+
+		private static void OnIsShadowEnabledChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var visualElement = bindable as FormsElement;
+			var enabled = (bool)newValue;
+			if (enabled)
+			{
+				visualElement.Effects.Add(new ShadowEffect());
+			}
+			else
+			{
+				foreach (var effect in visualElement.Effects)
+				{
+					if (effect is ShadowEffect)
+					{
+						visualElement.Effects.Remove(effect);
+						break;
+					}
+				}
+			}
+		}
+
+		public static bool GetIsShadowEnabled(BindableObject element)
+		{
+			return (bool)element.GetValue(IsShadowEnabledProperty);
+		}
+
+		public static void SetIsShadowEnabled(BindableObject element, bool value)
+		{
+			element.SetValue(IsShadowEnabledProperty, value);
+		}
+
+		public static bool GetIsShadowEnabled(this IPlatformElementConfiguration<iOS, FormsElement> config)
+		{
+			return GetIsShadowEnabled(config.Element);
+		}
+
+		public static IPlatformElementConfiguration<iOS, FormsElement> SetIsShadowEnabled(this IPlatformElementConfiguration<iOS, FormsElement> config, bool value)
+		{
+			SetIsShadowEnabled(config.Element, value);
+			return config;
+		}
+
+		public static readonly BindableProperty ShadowColorProperty =
+			BindableProperty.Create("ShadowColor", typeof(Color),
+			typeof(VisualElement), Color.Default);
+
+		public static Color GetShadowColor(BindableObject element)
+		{
+			return (Color)element.GetValue(ShadowColorProperty);
+		}
+
+		public static void SetShadowColor(BindableObject element, Color value)
+		{
+			element.SetValue(ShadowColorProperty, value);
+		}
+
+		public static Color GetShadowColor(this IPlatformElementConfiguration<iOS, Xamarin.Forms.VisualElement> config)
+		{
+			return GetShadowColor(config.Element);
+		}
+
+		public static IPlatformElementConfiguration<iOS, Xamarin.Forms.VisualElement> SetShadowColor(this IPlatformElementConfiguration<iOS, Xamarin.Forms.VisualElement> config, Color value)
```

Comment:
Convention is to use `FormsElement` instead of `Xamarin.Forms.VisualElement` here

Reply:
Thats the problem with using a template :P Will fix.